source/core/serial_solid.cc

```diff
@@ -479,21 +434,34 @@ SerialSolid<dim, spacedim>::write_output_results(
   std::shared_ptr<SimulationControl> simulation_control)
 {
   DataOut<dim, spacedim> data_out;
-  data_out.attach_dof_handler(solid_dh);
-
   data_out.attach_dof_handler(displacement_dh);
 
-  std::vector<std::string> solution_names(spacedim, "displacement");
-  std::vector<DataComponentInterpretation::DataComponentInterpretation>
-    data_component_interpretation(
-      spacedim, DataComponentInterpretation::component_is_part_of_vector);
+  {
+    std::vector<std::string> displacement_names(spacedim, "displacement");
+    std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      data_component_interpretation(
+        spacedim, DataComponentInterpretation::component_is_part_of_vector);
+
+
+    data_out.add_data_vector(displacement,
+                             displacement_names,
+                             DataOut<dim, spacedim>::type_dof_data,
+                             data_component_interpretation);
+  }
```

Comment:
Why those brackets? Why not just a short comment to separate the steps?

Reply:
Because otherwise I need to have two sets of variable with different names (the names, the interpretation, etc.), I felt it was a bit easier this way

Reply:
Oh I didn't know we should do that, nice